eva/udfs/decorators/decorators.py

```diff
@@ -54,8 +55,47 @@ def forward(input_signatures: List[IOArgument], output_signatures: List[IOArgume
 
     def inner_fn(arg_fn):
         def wrapper(*args):
+
+            # checking the user constraints
+            if check_input:
+                if len(input_signatures) > 0:
+                    args_lst = []
+                    if len(args) > 0:
+                        args_lst.append(args[0])
+
+                    for i, input_signature in enumerate(input_signatures):
+                        try:
+                            args_lst.append(
+                                # the first object in the forward function is self so we start from the second object
+                                input_signature.validate_object(args[i + 1], True)
+                            )
+                        except UDFIODefinitionError as e:
```

Comment:
I think it's better to use the assert style syntax that we have adopted in other parts of the code recently. @gaurav274 what do you think?

Reply:
since converting the shape or data type might throw an exception I am using the try-except